### PR TITLE
Convert t/mojo/delay.t to use subtests

### PR DESCRIPTION
**Pull request by dz-at-tc**
### Summary
Convert t/mojo/delay.t to use subtests.

### Motivation
One more file converted.

### References
Issue #1520

### CODE REVIEW

**Review by jhthorsen — APPROVED** [review]
Looks like some of the variables are not sorted alphabetically, but they weren't in the original commit either so I'm not going to request changes based on it.

https://github.com/mojolicious/mojo/pull/1568/files?w=1 looks good to me 👍

File: t/mojo/delay.t [hunk 1]
```diff
@@ -6,197 +6,211 @@ use Test::More;
...
+subtest 'Exception in last step' => sub {
+  my $failed = undef;
+  my $delay  = Mojo::IOLoop::Delay->new;
+  $delay->steps(sub { Mojo::IOLoop->next_tick(shift->begin) }, sub { die 'Last step!' })->catch(sub { $failed = pop })
+    ->wait;
+  like $failed, qr/^Last step!/, 'right error';
+};
+
+subtest 'Exception in second step' => sub {
+  my ($failed, $result) = ();
```
Comment (kraih) [inline t/mojo/delay.t:181]: These assignments do nothing.

File: t/mojo/delay.t [hunk 2]
```diff
@@ -6,197 +6,211 @@ use Test::More;
...
+    sub {
+      my ($delay, @numbers) = @_;
+      $result = \@numbers;
+    }
+  )->wait;
+  is_deeply $result, [2, 3, 2, 1, 4, 5], 'right results';
+};
+
+subtest 'One step' => sub {
+  my $result = undef;
```
Comment (kraih) [inline t/mojo/delay.t:76]: Assignment does nothing.

File: t/mojo/delay.t [hunk 3]
```diff
@@ -6,197 +6,211 @@ use Test::More;
...
+  is $result, undef, 'no result';
+  my $end = $delay->begin;
+  Mojo::IOLoop->next_tick(sub { $end->() });
+  is $result, undef, 'no result';
+  $delay->wait;
+  is $result, 1, 'right result';
+};
+
+subtest 'One step (reverse)' => sub {
+  my $result = undef;
```
Comment (kraih) [inline t/mojo/delay.t:91]: Same.

File: t/mojo/delay.t [hunk 4]
```diff
@@ -6,197 +6,211 @@ use Test::More;
...
+  my $end    = $delay->begin(0);
+  Mojo::IOLoop->next_tick(sub { $end->(23) });
+  $delay->steps(sub { $result = pop });
+  is $result, undef, 'no result';
+  $delay->wait;
+  is $result, 23, 'right result';
+};
+
+subtest 'End chain after first step' => sub {
+  my $result = undef;
```
Comment (kraih) [inline t/mojo/delay.t:102]: Same.

File: t/mojo/delay.t (hunk 1 above)
Comment (dz-at-tc) [reply t/mojo/delay.t:181]: Right, mea culpa, thanks for pointing these out!  I will fix this shortly...

**Comment by dz-at-tc** [comment]
A couple of follow up questions:

1) Should declarations like "my ($failed, $result);" be split into individual my declarations, one per line?
2) Should I alphabetize the var declarations as @jhthorsen pointed out above?

File: t/mojo/delay.t [hunk 5]
```diff
@@ -6,197 +6,211 @@ use Test::More;
...
+      my ($first, @numbers) = @_;
+      push @$result, @numbers;
+    }
+  )->wait;
+  is_deeply $result, [1, 2, 3, 2, 3, 2, 1, 4, 5, 6, 23, 24], 'right results';
+};
+
+subtest 'Exception in first step' => sub {
+  my $failed;
+  my $result;
```
Comment (kraih) [inline t/mojo/delay.t:165]: On separate lines is inconsistent.

Comment (dz-at-tc) [reply t/mojo/delay.t:165]: Very good, thanks for clarifying that the parentheses are preferred!

Comment (kraih) [reply t/mojo/delay.t:165]: The main preference is consistency. If something is unclear just look at how other files are formatted.

**Review by kraih — APPROVED** [review]

**Review by Grinnz — APPROVED** [review]

**Pull request merged**